server.go

```diff
@@ -1847,173 +1847,174 @@ func (s *server) Start() error {
 	cleanup := cleaner{}
 
 	s.start.Do(func() {
+		cleanup = cleanup.add(s.customMessageServer.Stop)
```

Comment:
this change is interesting...need to take a second look to see if this can be done safely, as there's some stop pattern that requires a certain state to be reached before it can stop, such as,
https://github.com/lightningnetwork/lnd/blob/acc595195cf8b08d100d1c360d53cbf9fe686867/peer/ping_manager.go#L201-L205

Reply:
I think that's fine, because we just call the stop function, and if we cannot stop it we just print a warning and continue with the cleanup of the other subsystems. We cannot know at which state the startup might fail so therefore I think we should precautiously call the stop function.

Reply:
I think YY is also referring to situations where we dont do nil checks (such as the above PingManager example) cause we just assume that Start has run & so assume that various variables have been initialised and are not nil.

Reply:
so as long as we are sure that either: 1) constructors take care of setting pointers or 2) the Stop funcs do nil checks where needed, then all should be good

Reply:
did some tests and ran into 1:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1019d87c4]

goroutine 1 [running]:
github.com/lightningnetwork/lnd/discovery.(*AuthenticatedGossiper).stop(0x14002290000)
        /Users/elle/LL/lnd/discovery/gossiper.go:756 +0x1a4
github.com/lightningnetwork/lnd/discovery.(*AuthenticatedGossiper).Stop.func1()
        /Users/elle/LL/lnd/discovery/gossiper.go:747 +0x1a4
sync.(*Once).doSlow(0x1400229000c, 0x14000b617f8)
```